src/libmongoc/tests/test-mongoc-primary-stepdown.c

```diff
@@ -54,9 +56,8 @@ _setup_test_with_client (mongoc_client_t *client)
 }
 
 static int
-_connection_count (mongoc_database_t *db)
+_connection_count (mongoc_client_t *client, uint32_t server_id)
```

Comment:
This was a fun bug. These tests check the connection count that a server reports to make assertions about whether connections are created or not. But, the checks must happen on the same server. It was relying on a primary read preference to select the same server after failover. But the primary read preference would only select the same server if monitoring had not yet discovered the newly changed primary. But with streamable ismaster, the failover is handled immediately and the new primary discovered, making the new connection count incorrect.

I strongly suspect this is also why this test flakes: https://jira.mongodb.org/browse/CDRIVER-3569

Fortunately Matt observed the same behavior in Node's tests and he brought this up in the drivers-streamable-ismaster channel.